projects/adc_challenge/src/main.c

```diff
@@ -0,0 +1,64 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// reads from the specified address and logs the value
+static void prv_read_from_adc_addr(const GpioAddress *address, void *context) {
+  GpioAddress *addr = context;
+  uint16_t adc_data = 0;
+  adc_read_converted_pin(*addr, &adc_data);
+
+  LOG_DEBUG("ADC pin has value %d\n", adc_data);
+}
+
+int main() {
+  interrupt_init();
+  gpio_init();
+  gpio_it_init();
+
+  GpioAddress input_addr = {
+    .port = GPIO_PORT_A,
+    .pin = 6,
+  };
+
+  GpioSettings input_settings = {
+    .direction = GPIO_DIR_IN,           //
```

Comment:
Do we need the `//` at the ends? I think i've seen this before but it seems unneseccary